### PR TITLE
Enable test `DataTest.printsString()`

### LINKED ISSUE #2590: DataTest.java:37-41: Enable the test {@link...
The puzzle `2437-f38310c3` from #2437 has to be resolved: 

https://github.com/objectionary/eo/blob/1325da69ced10e1fb5869424d08970c678848abb/eo-runtime/src/test/java/org/eolang/DataTest.java#L37-L41

The puzzle was created by rultor on 14-Nov-23. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by c71n93**
Closes #2590

---

## PR-Codex overview
This PR focuses on enabling the test `DataTest.printsString()` by updating the `toString()` method to return the expected string representation of data objects. 

### Detailed summary
- Enable the test `DataTest.printsString()`
- Update `toString()` method to return the expected string representation of data objects

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by c71n93** [comment]
@maxonfjvipon could you check this one, please?

File: eo-runtime/src/test/java/org/eolang/DataTest.java [hunk 1]
```diff
@@ -72,11 +67,18 @@ void printsEmptyByteArray() {
...
-            Matchers.containsString("Hello,\\nдруг!")
+        final byte[] input = "Hello,\nдруг!".getBytes(StandardCharsets.UTF_8);
+        final StringBuilder output = new StringBuilder(0);
+        for (final byte elem : input) {
+            if (output.length() > 0) {
+                output.append('-');
+            }
+            output.append(String.format("%02X", elem));
+        }
+        MatcherAssert.assertThat(
```
Comment (maxonfjvipon) [inline eo-runtime/src/test/java/org/eolang/DataTest.java:79]: @c71n93 let's add assertion message to the test

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]
@c71n93 good one! Just one comment from my side

File: eo-runtime/src/test/java/org/eolang/DataTest.java (hunk 1 above)
Comment (c71n93) [reply eo-runtime/src/test/java/org/eolang/DataTest.java:79]: @maxonfjvipon sorry, I don't understand. what kind of message?

Comment (maxonfjvipon) [reply eo-runtime/src/test/java/org/eolang/DataTest.java:79]: @c71n93 check [this](https://github.com/objectionary/eo/blob/69471fcc350ae14e415eac139b5795ef9bdb0883/eo-maven-plugin/src/test/java/org/eolang/maven/ParseMojoTest.java#L107) as an example

Comment (c71n93) [reply eo-runtime/src/test/java/org/eolang/DataTest.java:80]: @maxonfjvipon done

**Comment by c71n93** [comment]
@yegor256 could you check this one, please?

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**